packages/react/src/themes/teams/components/Button/buttonStyles.ts

```diff
@@ -1,215 +1,177 @@
-            color: textPrimaryColorHover,
+            color: v.textPrimaryColorHover,
           },
         }),
 
         ':focus': {
           boxShadow: 'none',
           outline: 'none',
-          ...(isFromKeyboard && borderFocusStyles),
+          ...(p.isFromKeyboard && borderFocusStyles),
         },
       }),
 
       // Overrides for "primary" buttons
-      ...(primary &&
-        !text && {
-          color: primaryColor,
-          backgroundColor: primaryBackgroundColor,
-          borderColor: primaryBorderColor,
+      ...(p.primary &&
+        !p.text && {
+          color: v.primaryColor,
+          backgroundColor: v.primaryBackgroundColor,
+          borderColor: v.primaryBorderColor,
 
           ':hover': {
-            color: primaryColorHover,
-            backgroundColor: primaryBackgroundColorHover,
+            color: v.primaryColorHover,
+            backgroundColor: v.primaryBackgroundColorHover,
           },
 
           ':focus': {
             boxShadow: 'none',
-            ...(isFromKeyboard
+            ...(p.isFromKeyboard
               ? {
-                  color: primaryColorFocus,
-                  backgroundColor: primaryBackgroundColorFocus,
+                  color: v.primaryColorFocus,
+                  backgroundColor: v.primaryBackgroundColorFocus,
                   ...borderFocusStyles,
                 }
-              : { ':active': { backgroundColor: primaryBackgroundColorActive } }),
+              : { ':active': { backgroundColor: v.primaryBackgroundColorActive } }),
           },
         }),
 
       // Overrides for "disabled" buttons
-      ...(disabled && {
+      ...(p.disabled && {
         cursor: 'default',
-        color: colorDisabled,
-        backgroundColor: backgroundColorDisabled,
-        borderColor: borderColorDisabled,
+        color: v.colorDisabled,
+        backgroundColor: v.backgroundColorDisabled,
+        borderColor: v.borderColorDisabled,
         boxShadow: 'none',
         ':hover': {
-          backgroundColor: backgroundColorDisabled,
+          backgroundColor: v.backgroundColorDisabled,
         },
       }),
 
-      ...(fluid && {
+      ...(p.fluid && {
         width: '100%',
         maxWidth: '100%',
       }),
 
-      ...(iconOnly && {
-        minWidth: height,
+      ...(p.size === 'small' && {
+        padding: v.sizeSmallPadding,
+        height: v.sizeSmallHeight,
+        minWidth: v.sizeSmallMinWidth,
+      }),
+
+      ...(p.iconOnly && {
+        minWidth: v.height,
         padding: 0,
       }),
     }
   },
 
   // modifies the text of the button
-  content: ({ variables }): ICSSInJSStyle => ({
+  content: ({ props: p, variables: v }): ICSSInJSStyle => ({
     overflow: 'hidden',
     textOverflow: 'ellipsis',
     whiteSpace: 'nowrap',
-    fontWeight: variables.contentFontWeight,
+    fontSize: v.fontSize,
+    fontWeight: v.contentFontWeight,
+
+    ...(p.size === 'small' && {
+      fontSize: v.sizeSmallFontSize,
```

Comment:
I think this need to have a line-height set for the small size and one for the default (medium) size to ensure that no descender are cut off...
![image](https://user-images.githubusercontent.com/828692/61852335-1b598d00-ae6e-11e9-824b-098cf471c00c.png)

Reply:
Adding the medium line-height as a default will also fix this - https://github.com/stardust-ui/react/issues/1376

Reply:
Cool, applied 👍